config/v3_0/types/storage_test.go

```diff
@@ -227,3 +234,94 @@ func TestStorageValidate(t *testing.T) {
 		}
 	}
 }
+
+func TestStorageValidateWarnings(t *testing.T) {
+	tests := []struct {
+		in  Storage
+		at  path.ContextPath
+		out error
+	}{
+		// test a disk with partitions with the same 'device' as a filesystem returns ErrPartitionsOverwritten
+		{
+			in: Storage{
+				Disks: []Disk{
+					{
+						Device: "/dev/sda",
+						Partitions: []Partition{
+							{}, {},
+						},
+					},
+				},
+				Filesystems: []Filesystem{
+					{
+						Device: "/dev/sda",
+					},
+				},
+			},
+			out: errors.ErrPartitionsOverwritten,
+			at:  path.New("", "filesystems", 0, "device"),
+		},
+		// test a disk with the same 'device' and 'WipeTable:=true' as a configured filesystem returns ErrFilesystemImplicitWipe
+		{
+			in: Storage{
+				Disks: []Disk{
+					{
+						Device:    "/dev/sda",
+						WipeTable: util.BoolToPtr(true),
+					},
+				},
+				Filesystems: []Filesystem{
+					{
+						Device:         "/dev/sda",
+						WipeFilesystem: util.BoolToPtr(false),
```

Comment:
This `WipeFilesystem` line and the next one are missing from the 3.3 and 3.4-exp tests.  I think it's actually okay to omit, since the field is more likely to be `nil` than `false`, but either way we should be consistent.

Reply:
I went and omit'd them. Nice catch :+1: